### ISSUE TITLE
Create new style or duplicate style (to create scenes with distinct rendering)

### ISSUE THREAD

**Issue by Fredosixx** [SketchUp/api-issue-tracker#1026]
I found **no way to duplicate a loaded style**, or to create a new one from scratch.

The use case is to be able to **create scenes with different rendering options**, noting that some rendering options, like `ModelTransparency`, are stored in styles.

Let's consider a new model with 2 scenes. It has one style (`SimpleStyle`). If you wish to make Scene #2 with XRay, while Scene #1 is kept not XRayed, then

**In the Sketchup GUI**, as soon as you switch to Xray and right-click to update the scene, you will get the prompt:

![image](https://github.com/user-attachments/assets/d5697f40-c83a-4afb-b65f-8970779c9b39)

1) If you select `Update the select style`, then too bad, Scene #1 will also become XRayed, because it uses this style too.

2) If you select 'Save as a new Style`, then a new style is created from the current style (so `SimpleStyle1`, which is what is wanted.

My question (or request) is that there seems to be no API equivalent to this last option 2). Furthermore, even if I was able to find the style source file (although there is no `style#path` method), I won't be able to load it, because it will either not load or override the existing style.

Basically, **this means that the API cannot be used to create a scene with Xray and another scene without XRay**.

**Note**: the workaround is to have my own styles, predefined with XRay and not XRay and load them. But this is not aht I was looking for.

**Comment by rwksict** [comment]
Thomas, thanks for logging this.  If I'd remembered this issue tracker in November, I'd have posted here then.  Thanks Fredo for getting it here
